src/parsec_client/operations/list_providers.md

```diff
@@ -1,10 +1,36 @@
 # ListProviders
 
-## Opcode: 8 (decimal), 0x0008 (hex)
+Gets a list of available Parsec providers to be used by clients. Opcode: 8 (`0x0008`)
 
-## Summary
+## Parameters
 
-Gets a list of available Parsec providers to be used by clients.
+No parameters are needed for this operation.
+
+## Results
+
+| Name        | Type                                           | Description                  |
+|-------------|------------------------------------------------|------------------------------|
+| `providers` | Vector of [`ProviderInfo`](#providerinfo-type) | List of provider information |
+
+### ProviderInfo type
+
+A `ProviderInfo` type contains the following members:
+
+| Name          | Type             | Description                                                               |
+|---------------|------------------|---------------------------------------------------------------------------|
+| `uuid`        | String           | Unique, permanent, identifier of the provider                             |
+| `description` | String           | Short description of the provider                                         |
+| `vendor`      | String           | Provider vendor                                                           |
+| `version_maj` | Unsigned integer | Provider implementation version major                                     |
+| `version_min` | Unsigned integer | Provider implementation version minor                                     |
+| `version_rev` | Unsigned integer | Provider implementation version revision number                           |
+| `id`          | Unsigned integer | Provider ID to use on the wire protocol to communicate with this provider |
```

Comment:
Are we still certain that we want to support at most one provider of each type, per service?

Reply:
I think this spec does not prevent it: currently our version of a "Provider ID" is a fixed number in an enum but we could expand it later to be dynamically allocated number and this spec won't change. I think that even if this number is dynamically allocated, it will still be called "Provider ID".

Reply:
True, I guess I was thinking of both UUID and ID at the same time!